packages/codegen/src/generateEntityCodegenFile.ts

```diff
@@ -238,6 +244,47 @@ export function generateEntityCodegenFile(config: Config, meta: EntityDbMetadata
   `;
 }
 
+function fieldHasDefaultValue(
+  config: Config,
+  meta: EntityDbMetadata,
+  field: EntityDbMetadata["primitives"] extends Array<infer U> ? U : never,
+): boolean {
+  let { fieldName, columnDefault, columnType } = field;
+
+  // if there's no default at all, return false
+  if (columnDefault === null) {
+    return false;
+  }
+
+  // even though default is defined as a number | boolean | string | null in reality pg-structure
+  // only ever returns a string | null so we make sure of that here
+  columnDefault = columnDefault.toString();
+
+  // if this value should be set elsewhere, return false
+  if (
+    ormMaintainedFields.includes(fieldName) ||
+    isDerived(config, meta.entity, fieldName) ||
+    isAsyncDerived(config, meta.entity, fieldName)
+  ) {
+    return false;
+  }
+
+  // try to validate that we actually got a primitive value and not arbitrary SQL
+  return (
```

Comment:
Wouldn't mind a unit test ... in the codegen directory itself? That threw some input at this and see what happens. Not a big deal given Joist's historical reliance on integration-tests for most...everything test related.

Reply:
since we don't have a `.test.ts` for this file yet it seems like making one would probably be enough work to be a PR all on it's own.  Since this PR is relatively simple i'd prefer to just rely on integration-tests.

Reply:
Sgtm.